teps/0091-trusted-resources.md

```diff
@@ -364,26 +372,137 @@ type KeylessRef struct {
 
 ### Configuration
 
-Reconciler will respond to verification failure based on ConfigMap settings to:
-1) Return error in reconciler, mark the status as error with `FailedVerification` reason;
-2) Not return error, but mark the TaskRun/PipelineRun as `FailedVerification`;
-3) Skip the verification;
-
-Configuration sample:
-```yaml
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  trusted-resources-verification-no-match-policy: "fail"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`trusted-resources-verification-no-match-policy`. (Optional, `ignore`, `warn` or `fail`, default to `ignore`):
+ * `ignore`: Don't fail the taskrun/pipelinerun and skip verification if no matching policies are found. Don't log.
+ * `warn`: Don't fail the taskrun/pipelinerun and log a warning if no matching policies are found.
+ * `fail`: Fail the taskrun/pipelinerun if no matching policies are found.
+
+**Note:** The current proposed `trusted-resources-verification-no-match-policy` will be added to replace the old `resource-verification-mode` in one release and this is not a backwards compatible change.
+
+### Condition Update
+
+Trusted resources should update the taskrun/pipelinerun’s condition to indicate if it passes verification or not. This can be done via knative api. Other options are listed in [alternatives](#alternatives).
```

Comment:
nit: 'via the knative api' sounds like we're including some knative specific functionality our API. afaik [this is a kubernetes specific convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties); knative pkg just provides us with libraries that make this easier. If I'm understanding correctly, it might be clearer here to refer to the k8s conventions instead of knative